### PR TITLE
support catkin_DEFINITIONS

### PR DESCRIPTION
**Pull request by mathias-luedtke**
Adds support for catkin_DEFINITIONS as discussed in https://github.com/ros/catkin/issues/610
The actual definitions has to be set via `cfg-extra` for now.

Should I add `add_defintions` to all `find_package(catkin..` occurrences in documentation?

### CODE REVIEW

File: doc/user_guide/find_package.rst [hunk 1]
```diff
@@ -14,13 +14,14 @@ simultaneously, use
                cpp_common rostime roscpp_traits
                roscpp_serialization sensor_msgs)
   include_directories(${catkin_INCLUDE_DIRS})
+  add_definitions(${catkin_DEFINITIONS})
 
   add_executable(myexec ...)
-  target_link_libraries(${catkin_LIBRARIES})
+  target_link_libraries(myexec ${catkin_LIBRARIES})
```
Comment (dirk-thomas) [inline doc/user_guide/find_package.rst:20]: Please create a separate PR for this fix (and the one below) since they can be directly merged.

**Comment by dirk-thomas** [comment]
Please see the [latest comment](https://github.com/ros/catkin/issues/610#issuecomment-270153665) on the referenced issue why simply concatenating exported definitions might not work well.

File: doc/user_guide/find_package.rst (hunk 1 above)
Comment (mathias-luedtke) [reply doc/user_guide/find_package.rst:20]: together with first commit

**Comment by mathias-luedtke** [comment]
closing for now.

**Pull request closed without merging**